### PR TITLE
[Backport stable/8.0] ci: merge deploy and auto-merge workflows into unified CI workflow

### PR DESCRIPTION
**Pull request by lenaschoenburg**
manual backport of #10616

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -413,3 +413,140 @@ jobs:
           name: Event File
           path: ${{ github.event_path }}
           retention-days: 1
+  deploy-snapshots:
```
Comment (megglos) [inline .github/workflows/ci.yml:416]: ❓ everything from here with the `github.ref == 'refs/heads/main'` conditions is just there for alignment? as effectively only auto-merge would be relevant. (Given dependabot creates PRs against stable branches)

Or should we adjust the snapshot steps to work for stable branches? e.g. pushing a 3.0.X-SNAPSHOT docker image?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -413,3 +413,140 @@ jobs:
...
+      - uses: ./.github/actions/build-zeebe
+        id: build-zeebe
+      - uses: ./.github/actions/build-docker
+        id: build-docker
+        with:
+          repository: camunda/zeebe
+          version: SNAPSHOT
+          push: true
+          distball: ${{ steps.build-zeebe.outputs.distball }}
+  notify-if-failed:
```
Comment (megglos) [inline .github/workflows/ci.yml:487]: ❓ hm might this actually also be useful to get notified on? so would we need to adjust it to run on a `stable` branch and report? But maybe not used and definitely out of scope as this didn't exist before anyway?

**Review by megglos — APPROVED** [review]
LGTM! Just some questions in regards to job setup only effective for the main branch.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (oleschoenburg) [reply .github/workflows/ci.yml:416]: Yeah, it's just for alignment. I think running this for branches other than main would be a separate topic and shouldn't stop us from backporting this.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (oleschoenburg) [reply .github/workflows/ci.yml:487]: Yeah definitely out of scope and not relevant until we actually run the deploy jobs on branches other than main.

**Pull request merged**